### ISSUE TITLE
Create required UI components for the front-end

### ISSUE THREAD

**Issue by blenderskool** [UdacityMobileWebScholarship/showcase-app#46]
**I'm submitting a ...**
- [ ] bug report
- [X] feature request

**Current behavior:**

As of now, most of the issues are open to other developers on **per-layout basis**. Since the app uses fundamental components throughout the app, every page may end up having different styles for every component which makes it really cumbersome to manage.

**Expected behavior:**

We can first create the **fundamental components required in the app**. This would allow other developers to join in and quickly create the layouts based on the desings. I am ready to start working on this :slightly_smiling_face:

**Comment by anurag-majumdar** [comment]
@blenderskool This is quite an important task indeed. Will you be carrying this feature out by yourself or you would like one more person to join in with you? If you need help I can assist you with these fundamental components. :smiley:

**Comment by blenderskool** [comment]
@anurag-majumdar Yes, even you can contribute. Based on some component guidelines that I have added to the albums, we can divide the components between ourselves.

**Comment by realslimshanky** [comment]
@blenderskool if this is the case, one has to create UI components. And then those components can be used by other contributors for layout development.
Inversely, since we have fixed views, those views can be designed individually to each other. Since we have very few time left till the end of the course we should have a working model. In future if we get time we can unify similar components.
What do you guys think?
@divyamrastogi @PrabhanshuAttri @palnabarun @shubhshrma @ParthS007 @ritikrishu

**Comment by divyamrastogi** [comment]
I actually agree with @blenderskool because it's not that hard to identify reusable components. Believe me, it saves a lot of time if you think ahead and plan these things a little.

In fact, even the layout can be a component since we've got a similar structure of the header section and the rest of the page. Let me identify these components and I'll create issues for them. Then, we can reuse those components to pick up issues created by @realslimshanky. What say?

**Comment by realslimshanky** [comment]
@divyamrastogi great. Thanks for taking such a huge task. Would help a lot in future processes.

**Comment by blenderskool** [comment]
@realslimshanky @divyamrastogi I have already added some in-depth component guidelines here https://photos.app.goo.gl/5nm1W01LSmGwOfGh2
Additional components can be created based on these principles

**Comment by anurag-majumdar** [comment]
Guys, before we dive in to the components part, we need to also add that we are changing the version from latest material-ui to the stable one 0.20. I am working on this part and will issue a PR soon.

The UI layouts are going to use 0.20 version of material-ui.

**Comment by anurag-majumdar** [comment]
Please ignore the last comment regarding 0.20. We are continuing with material-ui@next.
Are the reusable components in place? Then we can start building the pages asap.

@divyamrastogi @blenderskool @realslimshanky @PrabhanshuAttri @palnabarun
